integrations/pgvector/src/haystack_integrations/document_stores/pgvector/document_store.py

```diff
@@ -0,0 +1,380 @@
+
+        insert_statement = SQL("INSERT INTO {table_name} " + columns_str + " " + values_placeholder_str).format(
+            table_name=Identifier(self.table_name)
+        )
+
+        if policy == DuplicatePolicy.OVERWRITE:
+            update_statement = SQL(
+                "ON CONFLICT (id) DO UPDATE SET "
+                + ", ".join(f"{col} = EXCLUDED.{col}" for col in COLUMNS if col != "id")
+            )
+            insert_statement += update_statement
+        elif policy == DuplicatePolicy.SKIP:
+            insert_statement += SQL("ON CONFLICT DO NOTHING")
+
+        insert_statement += SQL(" RETURNING id")
+
+        try:
+            self._cursor.executemany(insert_statement, db_documents, returning=True)
+        except IntegrityError as ie:
+            self._connection.rollback()
+            raise DuplicateDocumentError from ie
+        except Error as e:
+            self._connection.rollback()
+            raise DocumentStoreError from e
+
+        # get the number of the inserted documents, inspired by psycopg3 docs
+        # https://www.psycopg.org/psycopg3/docs/api/cursors.html#psycopg.Cursor.executemany
+        written_docs = 0
+        while True:
+            if self._cursor.fetchone():
+                written_docs += 1
+            if not self._cursor.nextset():
+                break
+
+        return written_docs
+
+    def _from_haystack_to_pg_documents(self, documents: List[Document]) -> List[Dict[str, Any]]:
+        """
+        Internal method to convert a list of Haystack Documents to a list of dictionaries that can be used to insert
+        documents into the PgvectorDocumentStore.
+        """
+
+        db_documents = []
+        for document in documents:
+            db_document = {k: v for k, v in document.to_dict(flatten=False).items() if k not in ["score", "blob"]}
+
+            blob = document.blob
+            db_document["blob_data"] = blob.data if blob else None
+            db_document["blob_meta"] = Json(blob.meta) if blob and blob.meta else None
+            db_document["blob_mime_type"] = blob.mime_type if blob and blob.mime_type else None
+
+            db_document["dataframe"] = Json(db_document["dataframe"]) if db_document["dataframe"] else None
+            db_document["meta"] = Json(db_document["meta"])
+
+            db_documents.append(db_document)
+
+        return db_documents
+
+    def _from_pg_to_haystack_documents(self, documents: List[Dict[str, Any]]) -> List[Document]:
+        """
+        Internal method to convert a list of dictionaries from pgvector to a list of Haystack Documents.
+        """
+
+        haystack_documents = []
+        for document in documents:
+            haystack_dict = dict(document)
+            blob_data = haystack_dict.pop("blob_data")
+            blob_meta = haystack_dict.pop("blob_meta")
+            blob_mime_type = haystack_dict.pop("blob_mime_type")
+
+            if not haystack_dict["meta"]:
```

Comment:
`db_document["meta"]` contains `Json(...)`, what does `Document.meta` contain after `Document.from_dict`?

Reply:
sorry this condition is useless.
`haystack_dict['meta']` is always a valid dict and not None.
I'll remove this condition.